controlplane/kubeadm/controllers/controller.go

```diff
@@ -433,19 +433,19 @@ func (r *KubeadmControlPlaneReconciler) reconcileHealth(ctx context.Context, clu
 
 	// Ensure etcd is healthy
 	if err := r.managementCluster.TargetClusterEtcdIsHealthy(ctx, util.ObjectKey(cluster)); err != nil {
+		logger.V(2).Info("Waiting for control plane to pass etcd health check to continue reconciliation.", "cause", err)
+		r.recorder.Eventf(kcp, corev1.EventTypeWarning, "ControlPlaneUnhealthy",
+			"Waiting for control plane to pass etcd health check to continue reconciliation: %v", err)
 		// If there are any etcd members that do not have corresponding nodes, remove them from etcd and from the kubeadm configmap.
 		// This will solve issues related to manual control-plane machine deletion.
 		workloadCluster, err := r.managementCluster.GetWorkloadCluster(ctx, util.ObjectKey(cluster))
 		if err != nil {
-			return err
```

Comment:
Could we return an error aggregate instead of logging?

Reply:
I followed the convention used in other parts of the file. What did you have in mind?

Reply:
There are 2 messages logged in case of error. We could make them error and  combine them like this:
```
errList := []error{}
errList = append(errList, err)
return kerrors.NewAggregate(errList)
```

Reply:
But the calling function does not appear to log these errors: https://github.com/kubernetes-sigs/cluster-api/blob/master/controlplane/kubeadm/controllers/scale.go#L67-L69

Or am I missing something?

Reply:
We will see the errors in the log when reconcile return with errors:
`controller-runtime/controller "msg"="Reconciler error" "error"="Failed to get workload cluster client" "controller"="kubeadmcontrolplane" "name"="test" "namespace"="default"`

We try to avoid logging too much.

Reply:
I understand; however, please take a  look at reconcileHealth(), it returns https://github.com/kubernetes-sigs/cluster-api/blob/master/controlplane/kubeadm/controllers/controller.go#L431 and https://github.com/kubernetes-sigs/cluster-api/blob/master/controlplane/kubeadm/controllers/controller.go#L456

Additionally, as i pointed out above, the calling funcs do not log the error, but simply return a `capierrors.RequeueAfterError`: 
https://github.com/kubernetes-sigs/cluster-api/blob/master/controlplane/kubeadm/controllers/scale.go#L68
https://github.com/kubernetes-sigs/cluster-api/blob/master/controlplane/kubeadm/controllers/scale.go#L94

Unfortunately, I don't see a cleaner way to log these errors.

Reply:
We should remove `RequeueAfterError` and just return both a `reconcile.Result{}` and `error` in these functions.